TypedSyntax/src/show.jl

```diff
@@ -0,0 +1,201 @@
+#                 lastidx = last_byte(arg)
+#             end
+#         else
+#             error("unhandled head ", hd)
+#         end
+#     end
+#     # Print the closing ')' and the return type
+#     l = length(srctxt)
+#     lastidx += 1
+#     while lastidx <= l
+#         c = srctxt[lastidx]
+#         print(io, c)
+#         c == ')' && break
+#         lastidx = nextind(srctxt, lastidx)
+#     end
+#     print(io, "::")
+#     if iswarn
+#         printstyled(io, rt; color = is_type_unstable(rt) ? :red : :cyan)
+#     else
+#         print(io, rt)
+#     end
+#     return lastidx
+# end
+
+function show_src_expr(io::IO, node::TypedSyntaxNode, lastidx::Int; iswarn::Bool=false, hide_type_stable::Bool=false)
+    lastidx = catchup(io, node, lastidx)
+    _lastidx = last_byte(node)
+    if kind(node) == K"Identifier"
+        print(io, node.source[lastidx+1:_lastidx])
+        maybe_show_annotation(io, node.typ; iswarn, hide_type_stable)
+        return _lastidx
+    end
+    # We only handle "call" nodes. For anything else, just print the node (recursing into children)
+    if kind(node) != K"call"
+        for child in children(node)
+            lastidx = show_src_expr(io, child, lastidx; iswarn, hide_type_stable)
+        end
+        print(io, node.source[lastidx+1:_lastidx])
+        return _lastidx
+    end
+    pre = prepost = post = ""
+    if is_infix_op_call(node)   # wrap infix calls in parens before type-annotating
+        pre, post = "(", ")"
+        lastidx = catchup(io, first(children(node)), lastidx)
+    elseif is_prefix_op_call(node) # insert parens after prefix op and before type-annotating
+        prepost, post = "(", ")"
+    end
+    T = node.typ
+    # should we print a type-annotation?
+    type_annotate = isa(T, Vector{Int}) || (isa(T, Type) && (!hide_type_stable || is_type_unstable(T)))
+    type_annotate && print(io, pre)
+    for (childid, child) in enumerate(children(node))
+        childid == 2 && type_annotate && print(io, prepost)
+        lastidx = show_src_expr(io, child, lastidx; iswarn, hide_type_stable)
+    end
+    print(io, node.source[lastidx+1:_lastidx])
+    if type_annotate
+        show_annotation(io, T, post; iswarn)
+    end
+    return _lastidx
+end
+
+function maybe_show_annotation(io, @nospecialize(T); iswarn, hide_type_stable)
+    if isa(T, Type) && (!hide_type_stable || is_type_unstable(T))   # should we print a type-annotation?
+        show_annotation(io, T; iswarn)
+    end
+end
+
+function show_annotation(io, @nospecialize(T), post=""; iswarn::Bool)
+    print(io, post)
+    if isa(T, Vector{Int})
+        isempty(T) && return
+        if iswarn
+            printstyled(io, "::NF"; color=:yellow)
+        else
+            print(io, "::NF")
+        end
+        return
+    end
+    if iswarn
+        color = !is_type_unstable(T) ? :cyan :
+                 is_small_union(T) ? :yellow : :red
+        printstyled(io, "::", T; color)
+    else
+        print(io, "::", T)
```

Comment:
Should we still print this in cyan or perhaps just bold? Otherwise these might get confused for manual type annotations.